### PR TITLE
feat(ui): lineage - allow pipelines to be added for table to topic and vice versa

### PR DESCRIPTION
**Pull request by karanh37**
### Describe your changes:

Lineage - Allow pipelines to be added for table to topic and vice versa.
It also contains Lineage UI changes based on feedbacks.

I worked on ... because ...

https://github.com/open-metadata/OpenMetadata/assets/33024356/a7945437-4642-4906-a7e6-54ee8ccc5088

image

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomEdge.component.tsx [hunk 1]
```diff
@@ -87,12 +87,24 @@ export const CustomEdge = ({
...
-    return sourceType === EntityType.TABLE && targetType === EntityType.TABLE;
+  const isPipelineEdgeAllowed = (
+    sourceType: EntityType,
+    targetType: EntityType
+  ) => {
+    return (
+      (sourceType === EntityType.TABLE && targetType === EntityType.TABLE) ||
+      (sourceType === EntityType.TOPIC && targetType === EntityType.TABLE) ||
+      (sourceType === EntityType.TABLE && targetType === EntityType.TOPIC)
+    );
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomEdge.component.tsx:98]:
```suggestion
      return (
  [EntityType.TABLE, EntityType.TOPIC].indexOf(sourceType) > -1 &&
  [EntityType.TABLE, EntityType.TOPIC].indexOf(targetType) > -1
    );
```

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.component.tsx [hunk 2]
```diff
@@ -1654,9 +1653,7 @@ const EntityLineageComponent: FunctionComponent<EntityLineageProp> = ({
                 onOptionSelect={handleOptionSelect}
               />
             )}
-            {isEditMode && (
-              <Background gap={12} size={1} variant={BackgroundVariant.Lines} />
-            )}
+            <Background gap={12} size={1} />
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.component.tsx:1656]: Don't we need isEditMode condition here?

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/custom-node.less [hunk 3]
```diff
@@ -113,6 +123,23 @@
   top: 38px; // Need to show handles on top half
 }
 
+.react-flow .lineage-column-node-handle {
+  width: 20px;
+  border-radius: 0;
+  border-color: #0000001a;
+  background: #fff;
+  top: 0;
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/custom-node.less:131]: Did we remove box-shadow from all the buttons on the Lineage

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.component.tsx (hunk 2 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.component.tsx:1656]: Rupesh asked to keep background in both edit and non edit mode.

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/custom-node.less (hunk 3 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/custom-node.less:131]: Yes we have removed box-shadow

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**